diff --git a/src/AppBundle/Controller/ProductApiController.php b/src/AppBundle/Controller/ProductApiController.php
new file mode 100644
index 0000000..4ff272b
--- /dev/null
+++ b/src/AppBundle/Controller/ProductApiController.php
@@ -0,0 +1,18 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Service\ProductRepository;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+class ProductApiController extends Controller
+{
+    /**
+     * @Route("/api/products", name="api_product_list")
+     */
+    public function listAction()
+    {
+        return $this->json(['products' => 'foo']);
+    }
+}
\ No newline at end of file
