diff --git a/src/AppBundle/Controller/ProductApiController.php b/src/AppBundle/Controller/ProductApiController.php
index 4ff272b..0a1ec0a 100644
--- a/src/AppBundle/Controller/ProductApiController.php
+++ b/src/AppBundle/Controller/ProductApiController.php
@@ -13,6 +13,9 @@ class ProductApiController extends Controller
      */
     public function listAction()
     {
-        return $this->json(['products' => 'foo']);
+        $productRepo = new ProductRepository();
+        $products = $productRepo->findAll();
+
+        return $this->json($products);
     }
 }
\ No newline at end of file
