diff --git a/load_products.php b/load_products.php
deleted file mode 100644
index f5a5bc9..0000000
--- a/load_products.php
+++ /dev/null
@@ -1,51 +0,0 @@
-<?php
-
-use AppBundle\Entity\Category;
-use AppBundle\Entity\Product;
-use Doctrine\ORM\EntityManager;
-
-require __DIR__.'/vendor/autoload.php';
-
-$kernel = new AppKernel('dev', true);
-$kernel->boot();
-
-$container = $kernel->getContainer();
-
-$cat1 = new Category();
-$cat1->name = 'Electronics';
-
-$cat2 = new Category();
-$cat2->name = 'Baby';
-
-$cat3 = new Category();
-$cat3->name = 'Musical Instruments';
-
-$prod1 = new Product();
-$prod1->name = 'Amazon Echo';
-$prod1->price = 179.99;
-$prod1->description = 'Alexa... are you spying on me?';
-
-$prod2 = new Product();
-$prod2->name = 'Trombone';
-$prod2->price = 500;
-$prod2->description = 'The most excellent of the brass';
-
-$prod3 = new Product();
-$prod3->name = '500 Diapers';
-$prod3->price = 50;
-$prod3->description = '1 week\'s supply of diapers!';
-
-/** @var EntityManager $em */
-$em = $container->get('doctrine')->getManager();
-
-$em->createQuery('DELETE FROM AppBundle:Category')->execute();
-$em->createQuery('DELETE FROM AppBundle:Product')->execute();
-
-$em->persist($cat1);
-$em->persist($cat2);
-$em->persist($cat3);
-$em->persist($prod1);
-$em->persist($prod2);
-$em->persist($prod3);
-
-$em->flush();
diff --git a/src/AppBundle/Entity/Category.php b/src/AppBundle/Entity/Category.php
deleted file mode 100644
index 3fd5655..0000000
--- a/src/AppBundle/Entity/Category.php
+++ /dev/null
@@ -1,24 +0,0 @@
-<?php
-
-namespace AppBundle\Entity;
-
-use Doctrine\ORM\Mapping as ORM;
-
-/**
- * @ORM\Entity
- * @ORM\Table(name="category")
- */
-class Category
-{
-    /**
-     * @ORM\Id
-     * @ORM\GeneratedValue(strategy="AUTO")
-     * @ORM\Column(type="integer")
-     */
-    private $id;
-
-    /**
-     * @ORM\Column(type="string")
-     */
-    public $name;
-}
diff --git a/src/AppBundle/Entity/Product.php b/src/AppBundle/Entity/Product.php
deleted file mode 100644
index 0f3329a..0000000
--- a/src/AppBundle/Entity/Product.php
+++ /dev/null
@@ -1,34 +0,0 @@
-<?php
-
-namespace AppBundle\Entity;
-
-use Doctrine\ORM\Mapping as ORM;
-
-/**
- * @ORM\Entity
- * @ORM\Table(name="product")
- */
-class Product
-{
-    /**
-     * @ORM\Id
-     * @ORM\GeneratedValue(strategy="AUTO")
-     * @ORM\Column(type="integer")
-     */
-    private $id;
-
-    /**
-     * @ORM\Column(type="string")
-     */
-    public $name;
-
-    /**
-     * @ORM\Column(type="float")
-     */
-    public $price;
-
-    /**
-     * @ORM\Column(type="text")
-     */
-    public $description;
-}
