diff --git a/src/AppBundle/Controller/ProductApiController.php b/src/AppBundle/Controller/ProductApiController.php
index acc6dea..62f97fc 100644
--- a/src/AppBundle/Controller/ProductApiController.php
+++ b/src/AppBundle/Controller/ProductApiController.php
@@ -17,6 +17,9 @@ class ProductApiController extends Controller
         $productRepo = new ProductRepository();
         $products = $productRepo->findAll();
 
+        $this->container->get('logger')
+            ->debug(sprintf('Found %d products', count($products)));
+
         $json = $this->container->get('jms_serializer')
             ->serialize($products, 'json');
 
