diff --git a/assets/js/lib/uppercase_element.js b/assets/js/lib/uppercase_element.js
new file mode 100644
index 0000000..fbd2430
--- /dev/null
+++ b/assets/js/lib/uppercase_element.js
@@ -0,0 +1,3 @@
+module.exports = function($el) {
+    $el.addClass('text-uppercase');
+};
diff --git a/assets/js/main.js b/assets/js/main.js
index 9d0d2fc..7704b09 100644
--- a/assets/js/main.js
+++ b/assets/js/main.js
@@ -1,8 +1,9 @@
 const $ = require('jquery');
+const upperCaseElement = require('./lib/uppercase_element');
 require('../css/global.scss');
 
 $(document).ready(function() {
     $('h1').on('click', function(e) {
-        $(e.currentTarget).addClass('text-uppercase');
+        upperCaseElement($(e.currentTarget));
     });
 });
