diff --git a/src/AppBundle/Service/ProductRepository.php b/src/AppBundle/Service/ProductRepository.php
index 749308f..e4ddf8d 100644
--- a/src/AppBundle/Service/ProductRepository.php
+++ b/src/AppBundle/Service/ProductRepository.php
@@ -25,7 +25,7 @@ class ProductRepository
      * @param $id
      * @return Product|null
      */
-    public function findOne($id)
+    public function findOne($id) : ?Product
     {
         $pdo = $this->getPDO();
         $stmt = $pdo->prepare('SELECT * FROM product WHERE id = :id');
