diff --git a/src/AppBundle/Service/ProductRepository.php b/src/AppBundle/Service/ProductRepository.php
index a097e42..3acb10c 100644
--- a/src/AppBundle/Service/ProductRepository.php
+++ b/src/AppBundle/Service/ProductRepository.php
@@ -23,7 +23,7 @@ class ProductRepository
      * @param $id
      * @return Product|null
      */
-    public function findOne($id)
+    public function findOne($id) : ?Product
     {
         $pdo = $this->getPDO();
         $stmt = $pdo->prepare('SELECT * FROM product WHERE id = :id');
