diff --git a/src/AppBundle/Controller/ProductController.php b/src/AppBundle/Controller/ProductController.php
index 2d4e682..7b8f8ba 100644
--- a/src/AppBundle/Controller/ProductController.php
+++ b/src/AppBundle/Controller/ProductController.php
@@ -27,37 +27,11 @@ class ProductController extends Controller
      */
     public function showAction($id)
     {
-        $pdo = $this->getPDO();
-        $stmt = $pdo->prepare('SELECT * FROM product WHERE id = :id');
-        $stmt->execute(['id' => $id]);
-        $row = $stmt->fetch();
-
-        $product = $this->hydrateProduct($row);
+        $productRepo = new ProductRepository();
+        $product = $productRepo->findOne($id);
 
         return $this->render('product/show.html.twig', [
             'product' => $product
         ]);
     }
-
-    /**
-     * @return \PDO
-     */
-    private function getPDO()
-    {
-        $pdo = new \PDO('mysql:host=localhost;dbname=chicago_training', 'root');
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
-
-        return $pdo;
-    }
-
-    private function hydrateProduct(array $row)
-    {
-        $product = new Product();
-        $product->setId($row['id']);
-        $product->setName($row['name']);
-        $product->setPrice($row['price']);
-        $product->setDescription($row['description']);
-
-        return $product;
-    }
 }
\ No newline at end of file
diff --git a/src/AppBundle/Service/ProductRepository.php b/src/AppBundle/Service/ProductRepository.php
index 49fc9c9..156a3c9 100644
--- a/src/AppBundle/Service/ProductRepository.php
+++ b/src/AppBundle/Service/ProductRepository.php
@@ -22,6 +22,20 @@ class ProductRepository
     }
 
     /**
+     * @param $id
+     * @return Product
+     */
+    public function findOne($id)
+    {
+        $pdo = $this->getPDO();
+        $stmt = $pdo->prepare('SELECT * FROM product WHERE id = :id');
+        $stmt->execute(['id' => $id]);
+        $row = $stmt->fetch();
+
+        return $this->hydrateProduct($row);
+    }
+
+    /**
      * @return \PDO
      */
     private function getPDO()
