diff --git a/src/AppBundle/Controller/ProductController.php b/src/AppBundle/Controller/ProductController.php
index 34b729c..ac3c036 100644
--- a/src/AppBundle/Controller/ProductController.php
+++ b/src/AppBundle/Controller/ProductController.php
@@ -12,8 +12,7 @@ class ProductController extends Controller
      */
     public function indexAction()
     {
-        $pdo = new \PDO('mysql:host=localhost;dbname=chicago_training', 'root');
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+        $pdo = $this->getPDO();
 
         $results = $pdo->query('SELECT * FROM product');
 
@@ -27,8 +26,7 @@ class ProductController extends Controller
      */
     public function showAction($id)
     {
-        $pdo = new \PDO('mysql:host=localhost;dbname=chicago_training', 'root');
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+        $pdo = $this->getPDO();
         $stmt = $pdo->prepare('SELECT * FROM product WHERE id = :id');
         $stmt->execute(['id' => $id]);
         $product = $stmt->fetch();
@@ -37,4 +35,15 @@ class ProductController extends Controller
             'product' => $product
         ]);
     }
+
+    /**
+     * @return \PDO
+     */
+    private function getPDO()
+    {
+        $pdo = new \PDO('mysql:host=localhost;dbname=chicago_training', 'root');
+        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+
+        return $pdo;
+    }
 }
\ No newline at end of file
