diff --git a/app/config/services.yml b/app/config/services.yml
index 8a1b494..5d6ce48 100644
--- a/app/config/services.yml
+++ b/app/config/services.yml
@@ -16,3 +16,8 @@ services:
         calls:
             # $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
             - [ setAttribute, [3, 2] ]
+
+    app.product_serialization_subscriber:
+        class: AppBundle\Serializer\ProductSerializerSubscriber
+        tags:
+            - { name: jms_serializer.event_subscriber }
diff --git a/src/AppBundle/Serializer/ProductSerializerSubscriber.php b/src/AppBundle/Serializer/ProductSerializerSubscriber.php
new file mode 100644
index 0000000..e8917e0
--- /dev/null
+++ b/src/AppBundle/Serializer/ProductSerializerSubscriber.php
@@ -0,0 +1,29 @@
+<?php
+
+namespace AppBundle\Serializer;
+
+use AppBundle\Model\Product;
+use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
+use JMS\Serializer\EventDispatcher\ObjectEvent;
+use JMS\Serializer\JsonSerializationVisitor;
+
+class ProductSerializerSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+            array(
+                'event' => 'serializer.post_serialize',
+                'method' => 'onPostSerialize',
+                'class' => Product::class,
+            ),
+        );
+    }
+
+    public function onPostSerialize(ObjectEvent $event)
+    {
+        /** @var JsonSerializationVisitor $visitor */
+        $visitor = $event->getVisitor();
+        $visitor->setData('salesPrice', 1000);
+    }
+}
\ No newline at end of file
