diff --git a/src/AppBundle/Controller/LuckyController.php b/src/AppBundle/Controller/LuckyController.php
index 7cd8768..d00cece 100644
--- a/src/AppBundle/Controller/LuckyController.php
+++ b/src/AppBundle/Controller/LuckyController.php
@@ -8,11 +8,11 @@ use Symfony\Component\HttpFoundation\Response;
 class LuckyController
 {
     /**
-     * @Route("/lucky/number")
+     * @Route("/lucky/number/{max}")
      */
-    public function numberAction()
+    public function numberAction($max)
     {
-        $number = mt_rand(0, 100);
+        $number = mt_rand(0, $max);
 
         return new Response(
             '<html><body>Lucky number: '.$number.'</body></html>'
