diff --git a/src/AppBundle/Service/DiscountManager.php b/src/AppBundle/Service/DiscountManager.php
index 423e10d..52c37f2 100644
--- a/src/AppBundle/Service/DiscountManager.php
+++ b/src/AppBundle/Service/DiscountManager.php
@@ -16,6 +16,10 @@ class DiscountManager
             return $product->getPrice() * .8;
         }
 
+        if ($product->getPrice() == 500) {
+            return 250;
+        }
+
         return $product->getPrice() * .7;
     }
 }
\ No newline at end of file
diff --git a/tests/AppBundle/Service/DiscountManager.php b/tests/AppBundle/Service/DiscountManager.php
index 3da7e83..c4a53c5 100644
--- a/tests/AppBundle/Service/DiscountManager.php
+++ b/tests/AppBundle/Service/DiscountManager.php
@@ -27,6 +27,7 @@ class DiscountManagerTest extends TestCase
         $tests[] = [30, 27];
         $tests[] = [120, 96];
         $tests[] = [200, 140];
+        $tests[] = [500, 250];
 
         return $tests;
     }
