diff --git a/src/AppBundle/Service/DiscountManager.php b/src/AppBundle/Service/DiscountManager.php
index f2905e9..423e10d 100644
--- a/src/AppBundle/Service/DiscountManager.php
+++ b/src/AppBundle/Service/DiscountManager.php
@@ -8,6 +8,14 @@ class DiscountManager
 {
     public function getDiscountedPrice(Product $product)
     {
-        return 27;
+        if ($product->getPrice() < 100) {
+            return $product->getPrice() * .9;
+        }
+
+        if ($product->getPrice() < 200) {
+            return $product->getPrice() * .8;
+        }
+
+        return $product->getPrice() * .7;
     }
 }
\ No newline at end of file
diff --git a/tests/AppBundle/Service/DiscountManager.php b/tests/AppBundle/Service/DiscountManager.php
index 758af39..3da7e83 100644
--- a/tests/AppBundle/Service/DiscountManager.php
+++ b/tests/AppBundle/Service/DiscountManager.php
@@ -8,13 +8,26 @@ use PHPUnit\Framework\TestCase;
 
 class DiscountManagerTest extends TestCase
 {
-    public function testGetDiscountedPrice()
+    /**
+     * @dataProvider getPriceTests
+     */
+    public function testGetDiscountedPrice($productPrice, $expectedDiscountedPrice)
     {
         $discountManager = new DiscountManager();
 
         $product = new Product();
-        $product->setPrice(30);
+        $product->setPrice($productPrice);
 
-        $this->assertEquals(27, $discountManager->getDiscountedPrice($product));
+        $this->assertEquals($expectedDiscountedPrice, $discountManager->getDiscountedPrice($product));
+    }
+
+    public function getPriceTests()
+    {
+        $tests = [];
+        $tests[] = [30, 27];
+        $tests[] = [120, 96];
+        $tests[] = [200, 140];
+
+        return $tests;
     }
 }
\ No newline at end of file
