diff --git a/tests/AppBundle/Controller/DefaultControllerTest.php b/tests/AppBundle/Controller/DefaultControllerTest.php
index 594803c..32326b4 100644
--- a/tests/AppBundle/Controller/DefaultControllerTest.php
+++ b/tests/AppBundle/Controller/DefaultControllerTest.php
@@ -15,4 +15,16 @@ class DefaultControllerTest extends WebTestCase
         $this->assertEquals(200, $client->getResponse()->getStatusCode());
         $this->assertContains('Welcome to Symfony', $crawler->filter('#container h1')->text());
     }
+
+    public function testProductList()
+    {
+        $client = static::createClient();
+
+        $crawler = $client->request('GET', '/products');
+
+        $this->assertEquals(200, $client->getResponse()->getStatusCode());
+        $link = $crawler->filter('table a')->link();
+        $client->click($link);
+        $this->assertEquals(200, $client->getResponse()->getStatusCode());
+    }
 }
