diff --git a/tests/AppBundle/Controller/ProductApiControllerTest.php b/tests/AppBundle/Controller/ProductApiControllerTest.php
new file mode 100644
index 0000000..ae56dcb
--- /dev/null
+++ b/tests/AppBundle/Controller/ProductApiControllerTest.php
@@ -0,0 +1,13 @@
+<?php
+
+namespace Tests\AppBundle\Controller;
+
+use PHPUnit\Framework\TestCase;
+
+class ProductApiControllerTest extends TestCase
+{
+    public function testPOSTCreateProduct()
+    {
+        $this->assertEquals(42, 21*2);
+    }
+}
\ No newline at end of file
